src/clj/puppetlabs/puppetserver/shell_utils.clj

```diff
@@ -39,9 +39,11 @@
 
 (schema/defn ^:always-validate java-exe-options :- ShellUtils$ExecutionOptions
   [{:keys [env in]} :- ExecutionOptions]
-  (doto (ShellUtils$ExecutionOptions.)
-    (.setEnv (if env (ks/mapkeys name env) {}))
-    (.setStdin in)))
+  (let [exe-options (ShellUtils$ExecutionOptions.)]
+    (.setStdin exe-options in)
+    (when env
+      (.setEnv exe-options (ks/mapkeys name env)))
```

Comment:
Ah, because before this would have been passed through to Java as nil/null when no environment variables were set but now it would be an empty map, which would cause the Apache library to spawn the command with an empty environment?

Just curious, what does the ExecutionStub do when running under an MRI Ruby master?  Did it allow all environment variables from the original process to be passed through, only allow a whitelist of some common ones to be passed through, or something else?